R/engine.R

```diff
@@ -174,6 +174,60 @@ get_engine_opts = function(opts, engine, fallback = '') {
 
 get_engine_path = function(path, engine) get_engine_opts(path, engine, engine)
 
+# engine.opts = list(command, file, args, args1, args2)
+eng_exec = function(options) {
+  opts = options$engine.opts %n% options[c('command', 'file', 'ext', 'args', 'args1', 'args2')]
+  if (!is.character(cmd <- opts$command)) stop(
+    "The command of the 'exec' engine was not specified."
+  )
+
+  # default options
+  opts2 = list(
+    ext = identity, file = function(code, file) {
+      write_utf8(code, file)
+      file
+    }, args = function(code, file) {
+      file
+    }, clean = function(file) {
+      unlink(file)
+    }
+  )
+
+  opts = merge_list(opts2, opts)
+  cmd2 = basename(cmd)  # in case command is a full path
+  ext = opts$ext(cmd2)  # file extension
+  f = if (is.function(opts$file)) {
+    f = wd_tempfile(cmd2, paste0('.', ext))
+    if (is.function(opts$clean)) on.exit(opts$clean(f), add = TRUE)
+    opts$file(options$code, f)
+  } else if (is.character(opts$file) {
```

Comment:
```suggestion
  } else if (is.character(opts$file)) {
```

Sorry missing parenthesis